src/Command/MergeAssetsAndVariationsFilesCommand.php

```diff
@@ -0,0 +1,252 @@
+    }
+
+    private function mergeFiles(OutputInterface $output, string $targetFilePath)
+    {
+        $progressBar = new ProgressBar($output, $this->assetsReader->count());
+        $progressBar->start();
+
+        /** @var Writer $writer */
+        $writer = WriterFactory::create(Type::CSV);
+        $writer->setFieldDelimiter(self::CSV_FIELD_DELIMITER);
+        $writer->setFieldEnclosure(self::CSV_FIELD_ENCLOSURE);
+        $writer->openToFile($targetFilePath);
+        $writer->addRow($this->getAssetManagerFileHeaders());
+
+        foreach ($this->assetsReader as $assetLineNumber => $row) {
+            if ($assetLineNumber === 1) {
+                continue;
+            }
+
+            if (!$this->isHeaderValid($this->assetsReader, $row)) {
+                continue;
+            }
+
+            $assetLine = array_combine($this->assetsReader->getHeaders(), $row);
+
+            $variationsForThisAsset = [];
+            foreach ($this->variationsReader as $variationLineNumber => $variationRow) {
+                if ($variationLineNumber === 1) {
+                    continue;
+                }
+
+                if (!$this->isHeaderValid($this->variationsReader, $variationRow)) {
+                    continue;
+                }
+
+                $variationLine = array_combine($this->variationsReader->getHeaders(), $variationRow);
+
+                if ($variationLine['asset'] === $assetLine['code']) {
+                    $variationsForThisAsset[] = $variationLine;
+                }
+            }
+
+            $newAsset = $this->mergeAssetAndVariations($assetLine, $variationsForThisAsset);
+
+            $writer->addRow($newAsset);
+            $progressBar->advance();
+        }
+
+        $writer->close();
+        $progressBar->finish();
+    }
+
+    private function isHeaderValid(CsvReader $reader, $row)
+    {
+        return count($reader->getHeaders()) === count($row);
+    }
+
+    /**
+     * The new file Headers are a merge between the Asset CSV file headers and the generated value keys
+     * computed from the Variation CSV file headers (attribute-channel-locale)
+     */
+    private function getAssetManagerFileHeaders(): array
+    {
+        $assetHeaders = $this->assetsReader->getHeaders();
+        $valuesHeaders = [];
+        foreach ($this->channels as $channel) {
+            $valuesHeaders[] = sprintf('reference_file-%s', $channel);
+            $valuesHeaders[] = sprintf('variation_file-%s', $channel);
+
+            foreach ($this->locales as $locale) {
+                $valuesHeaders[] = sprintf('reference_file-%s-%s', $channel, $locale);
+                $valuesHeaders[] = sprintf('variation_file-%s-%s', $channel, $locale);
+            }
+        }
+
+        return array_merge($assetHeaders, $valuesHeaders);
+    }
+
+    /**
+     * Retrieve Channels & Locales we'll need to write the new Assets values for.
+     */
+    private function retrieveChannelsAndLocales(): void
```

Comment:
I would have fetched them from the API, but it's maybe better this way.

Reply:
Indeed we could, we'll only have to replace this method if we do this